### PR TITLE
Fix single options argument case

### LINKED ISSUE #8: Broken tests & failing in browsers
After 1.1.0
```js
isMobile({tablet: true})
```
fails in browser, as well as assert#8 in test.

### PR DESCRIPTION
**Pull request by dy**
Fixes #8 and broken browser test now.
@juliangruber please review.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
   },
   "dependencies": {},
   "devDependencies": {
-    "tape": "~1.0.4"
+    "tape": "^4.9.1"
```
Comment (dy) [inline package.json]: Since now there are two test clauses, running old `tape.only` still runs first clause.

File: test.js [hunk 2]
```diff
@@ -20,3 +20,21 @@ test('is mobile', function (t) {
   t.end();
 });
 
+test('mock window.navigator', function (t) {
+  if (typeof document !== 'undefined') return t.end();
```
Comment (dy) [inline test.js]: In browser skips that test, because userAgent is emulated via devtools

File: index.js [hunk 3]
```diff
@@ -8,11 +8,11 @@ var mobileRE = /(android|bb\d+|meego).+mobile|avantgo|bada\/|blackberry|blazer|c
 var tabletRE = /(android|bb\d+|meego).+mobile|avantgo|bada\/|blackberry|blazer|compal|elaine|fennec|hiptop|iemobile|ip(hone|od)|iris|kindle|lge |maemo|midp|mmp|mobile.+firefox|netfront|opera m(ob|in)i|palm( os)?|phone|p(ixi|re)\/|plucker|pocket|psp|series(4|6)0|symbian|treo|up\.(browser|link)|vodafone|wap|windows (ce|phone)|xda|xiino|android|ipad|playbook|silk/i;
 
 function isMobile (ua, opts) {
-  if (!opts && ua !== null && typeof ua === 'object') opts = ua;
-  if (!ua && typeof navigator !== 'undefined') ua = navigator.userAgent;
-  if (ua && ua.headers && typeof ua.headers['user-agent'] === 'string') {
-    ua = ua.headers['user-agent'];
+  if (ua && typeof ua === 'object') {
+    if (!opts) opts = ua;
+    ua = ua.headers && ua.headers['user-agent'];
```
Comment (dy) [inline index.js]: If single object passed as an argument, that is more explicitly considered both as UA object and options.

**Comment by dy** [comment]
@juliangruber any reaction?

**Comment by juliangruber** [comment]
What value does `window.navigator` have in your no-browser case? I don't yet understand your situation fully.

**Comment by dy** [comment]
@juliangruber I mean I can write something cross-platform like 
```js
isMobile(global.navigator && global.navigator.userAgent, {tablet: true})
```
but that looks cumbersome compared to
```js
isMobile({tablet: true})
```
which is also expected API, isn't that?

**Comment by juliangruber** [comment]
https://github.com/juliangruber/is-mobile/pull/10

**Comment by dy** [comment]
Covered by #10

**Pull request closed without merging**